instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/config.go

```diff
@@ -19,29 +19,31 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
-type config struct {
-	TracerProvider    trace.TracerProvider
-	TextMapPropagator propagation.TextMapPropagator
-	AttributeSetter   []AttributeSetter
+// Config holds configuration details for the SDK.
+type Config struct {
```

Comment:
Why is config becoming public?

Reply:
`AttributeSetter` is a public type with a shape of `func(context.Context, middleware.InitializeInput, *Config)`.

Reply:
That seems like a smell, as it allows anybody to update the Config pointer with `WithAttributeSetter`...

Reply:
But why would anyone do that in the first place? I mean, we can pass the value instead of course, but why would end users compromise the config?

Reply:
It's not just about end users being able to compromise the config. It's about providing them with an elegant and coherent interface.

Reply:
Here I agree. I don't really like the idea of changing the interface (and making it not backward compatible) myself, yet I would not like to pollute the context too, but that seems like the only easy way to not expose unnecessary details or sacrifice exportable setters. Do you maybe have a vision for this?

Reply:
@dmathieu I committed the solution with the config passed via the context, which kinda makes the config not a part of  the public API, which on the other hand makes using stuff like `RecordSNSPhoneNumber` completely internal. I guess that's more or less a valid compromise as in case of custom attribute setters the users can control the behavior of the setters themselves and they don't need to access the config to do that.

Reply:
You may have forgotten to push. But if you were passing the config through context's values, it wouldn't be a good solution either.

I'd be interested to hear other people's ideas here. But one solution could be to have two types of attribute setters. Private ones, which do get the config and are defined by this module.
And public ones, which can be set externally by users, but don't get access to the config.